### PR TITLE
Addressing #26294

### PR DESCRIPTION
**Pull request by hanswenzel**
This release contains the following changes:

fcl files have been repaired to use new ParticleListAction defaults and to use larg4SingleGen. The routine also
demonstrates how to access the MCTruth and MCParticle data objects:
modified:   fcl/testmctruth.fcl : enable dropping of em shower particles, use simpleLArTPC.gdml
            ParticleListAction_service.cxx: keep track of trackids that were dropped and add set of dropped track Ids to the event.
	                                    if the track ID is on the set no MCParticle has been produced.
					    No trajectory points ar added to MCTrajectory for steps that were limited by the StepLimiter
					    process in the active liquid Argon.
            CheckMCParticle_module.cc: example how to access the set of dropped particles.

new: simpleLArTPC.gdml : simple liquid Argon TPC example use copynumber to uniquely identify Sensitive detectors.

### CODE REVIEW

**Comment by lgarren** [comment]
trigger build

File: larg4/Analysis/CheckMCParticle_module.cc [hunk 1]
```diff
@@ -56,19 +56,43 @@ void larg4::CheckMCParticle::beginJob()
...
+  auto allDropped =  event.getMany<std::set<int>>();
+  for (auto const& dropped : allDropped) {
+    
+    std::cout << "List of dropped Tracks: "<<std::endl;
+    for(auto const& droppedTrack :  *dropped )
+      {
+	std::cout << droppedTrack << " ";
+      }
+    std::cout <<std::endl;   
+  }
```
Comment (knoepfel) [inline larg4/Analysis/CheckMCParticle_module.cc:69]: Suggest wrapping this printout in an `#if defined _verbose_` block, similar to what is done below.

File: larg4/pluginActions/ParticleListAction_service.cxx [hunk 2]
```diff
@@ -563,21 +569,29 @@ namespace larg4 {
...
+    //  {
+    //	std::cout <<"StepLimit ********************* "<<std::endl;
+    //  }
+   
+    // mf::LogDebug("ParticleListActionService::SteppingAction")
     /*
-    mf::LogDebug("ParticleListActionService::SteppingAction")
-    << ": DEBUG - process='"
+    std::cout
+    << ": DEBUG - process="
```
Comment (knoepfel) [inline larg4/pluginActions/ParticleListAction_service.cxx]: I would restore the `mf::LogDebug` statement here.

File: larg4/pluginActions/ParticleListAction_service.cxx [hunk 3]
```diff
@@ -563,21 +569,29 @@ namespace larg4 {
...
+    // change below:
+    // This method is being called for every step that
     // the track passes through, but we don't want to update the
     // trajectory information if we're just updating voxels. To check
     // for this, look at the process name for the step, and compare it
     // against the voxelization process name (set in PhysicsList.cxx).
     G4String process = step->GetPostStepPoint()->GetProcessDefinedStep()->GetProcessName();
-    G4bool ignoreProcess = process.contains("LArVoxel") || process.contains("OpDetReadout");
-
+    G4bool ignoreProcess = process.contains("StepLimiter");
```
Comment (knoepfel) [inline larg4/pluginActions/ParticleListAction_service.cxx:575]: Is this change related to issue #26924?

File: larg4/pluginActions/ParticleListAction_service.cxx [hunk 4]
```diff
@@ -338,9 +342,9 @@ namespace larg4 {
           // figure out the ultimate parentage of this particle
           // first add this track id and its parent to the fParentIDMap
           fParentIDMap[trackID] = parentID;
-
-          fCurrentTrackID = -1 * this->GetParentage(trackID);
-
+	  fCurrentTrackID = this->GetParentage(trackID);
```
Comment (knoepfel) [inline larg4/pluginActions/ParticleListAction_service.cxx:345]: The -1 has been removed here but kept below on line 374.  Is that intentional?

**Review by knoepfel — COMMENTED** [review]
Thanks, @hanswenzel, for the PR.  I have some questions for you below.

File: larg4/Analysis/CheckMCParticle_module.cc (hunk 1 above)
Comment (hanswenzel) [reply larg4/Analysis/CheckMCParticle_module.cc:69]: Hi Kyle, 
done. 
thanks for the comments. Should have said that I am planning for a cleaned up version for next week.

File: larg4/pluginActions/ParticleListAction_service.cxx (hunk 2 above)
Comment (hanswenzel) [reply larg4/pluginActions/ParticleListAction_service.cxx]: done

File: larg4/pluginActions/ParticleListAction_service.cxx (hunk 3 above)
Comment (hanswenzel) [reply larg4/pluginActions/ParticleListAction_service.cxx:575]: Basically it's a bug fix the LArVoxel read out has been replaced by the StepLimiter so there are no LArVoxel or OpDetReadout processes in the new system. 
In some way it is related to #26924. It will reduce the memory foot print since less step points are added to the MCTrajectories.

File: larg4/pluginActions/ParticleListAction_service.cxx (hunk 4 above)
Comment (hanswenzel) [reply larg4/pluginActions/ParticleListAction_service.cxx:345]: I got rid of the -1 everywhere it doesn't matter and ID always should be positive  (usually I enforce it by using unsigned int)

**Comment by lgarren** [comment]
trigger build

File: larg4/pluginActions/ParticleListAction_service.cxx (hunk 3 above)
Comment (knoepfel) [reply larg4/pluginActions/ParticleListAction_service.cxx:575]: Understood

**Comment by knoepfel** [comment]
Thanks @hanswenzel for the explanations.  Will approve after L1 approval.

**Comment by lgarren** [comment]
approve
